MailCore/Cache/Attachments/AttachmentsManagerWorker/AttachmentsManagerWorker.swift

```diff
@@ -384,30 +395,62 @@ extension AttachmentsManagerWorker: AttachmentsManagerWorkable {
                 // Title if any usable
                 var modified = false
                 if draftInContext.subject.isEmpty,
-                   let title = textAttachment.title {
-                    draftInContext.subject = title
+                   !anyUsableTitle.isEmpty {
+                    draftInContext.subject = anyUsableTitle
                     modified = true
                 }
 
-                // Url if any after a minimalistic input sanitising
-                if let body = textAttachment.body,
-                   let bodyUrl = URL(string: body),
-                   !body.isEmpty {
-                    draftInContext.body = "<div><a href=\"\(bodyUrl.absoluteString)\">"
-                        + bodyUrl.absoluteString
-                        + "</a></div>"
-                        + draftInContext.body
+                if !formattedBodyUrls.isEmpty {
+                    draftInContext.body = formattedBodyUrls + draftInContext.body
                     modified = true
                 }
 
                 guard modified else {
                     return
                 }
+
                 realm.add(draftInContext, update: .modified)
             }
         }
     }
 
+    private func anyUsableTitle(in textAttachments: [TextAttachment]) -> String {
+        textAttachments.first { textAttachment in
+            guard let title = textAttachment.title,
+                  !title.isEmpty else {
+                return false
+            }
+
+            return true
+        }?.title ?? ""
```

Comment:
Maybe it's juste me but I find this more readable than a big trailing closure with guard ?
```suggestion
        textAttachments.first { $0.title?.isEmpty == false }?.title ?? ""
```

Reply:
To me longer can be easier to read, as less things are processed per line, but that's fine with me.